### PR TITLE
Migrate kill command to cobra

### PR DESCRIPTION
**Pull request by WeiZhang555**
Part of #23211

Signed-off-by: Zhang Wei zhangwei555@huawei.com

### CODE REVIEW

**Comment by thaJeztah** [comment]
oh! needs a rebase (sorry, lots of activity :smile:)

**Comment by WeiZhang555** [comment]
@thaJeztah Yep, it's true :laughing:  Rebased.

File: api/client/container/kill.go [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+	flags := cmd.Flags()
+	flags.StringVarP(&opts.signal, "signal", "s", "KILL", "Signal to send to the container")
+	return cmd
+}
+
+func runKill(dockerCli *client.DockerCli, opts *killOptions) error {
+	var errs []string
+	for _, name := range opts.containers {
+		if err := dockerCli.Client().ContainerKill(context.Background(), name, opts.signal); err != nil {
```
Comment (vdemeester) [inline api/client/container/kill.go]: Ah, I forgot this one when updating context use…
@WeiZhang555 could you update this to share the context ? something like 

```go
ctx := context.Background()
for _, name := range opts.containers {
    if err := dockerCli.Client().ContainerKill(ctx, …) // …
}
```

Comment (WeiZhang555) [inline api/client/container/kill.go]: Oh, I see. 
I'll do that.

**Comment by WeiZhang555** [comment]
@vdemeester Updated as you suggested.

**Comment by vdemeester** [comment]
LGTM 🐰

**Comment by thaJeztah** [comment]
LGTM!

**Pull request merged**